pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchgit
+, avahi
+, boost
+, cmake
+, cudaPackages
+, eigen
+, ffmpeg
+, freetype
+, git
+, glm
+, glslang
+, harfbuzz
+, libdrm
+, libva
+, libpulseaudio
+, monado
+, nlohmann_json
+, onnxruntime
+, openxr-loader
+, pkg-config
+, python3
+, shaderc
+, spdlog
+, systemd
+, udev
+, vulkan-headers
+, vulkan-loader
+, vulkan-tools
+, x264
+, xorg
+}:
+let
+  vendorMonado = monado.overrideAttrs rec {
+    # Version stated in CMakeList for WiVRn 0.11
+    version = "57e937383967c7e7b38b5de71297c8f537a2489d";
+
+    src = fetchgit {
+      url = "https://gitlab.freedesktop.org/monado/monado.git";
+      rev = "${version}";
+      hash = "sha256-O/Td2WccTr4Fa8U64/lVfnidSIH5t3gWuFXCsEVf7bk=";
+    };
+
+    postInstall = ''
+      mv src/xrt/compositor/libcomp_main.a $out/lib/libcomp_main.a
+    '';
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "wivrn";
+  version = "0.11";
+
+  src = fetchFromGitHub {
+    owner = "meumeu";
+    repo = "wivrn";
+    rev = "v${version}";
+    hash = "sha256-E48JYhkBVZfb7S/FW0F8RbMtx4GIJwfXfs4KAF3gn8A=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    cudaPackages.cuda_nvcc
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    avahi
+    boost
+    cudaPackages.cuda_cudart
+    eigen
+    ffmpeg
+    freetype
+    git
```

Comment:
Just checking: does it really use git after the build/at runtime?

Reply:
At some point when making the package, I got an error about missing git but I tested it and it doesn't seem to be used so I'll remove it.

Reply:
👍🏻 I see. But to reiterate, it's the same logic as with `cuda_nvcc`: if it's executed at build time on the build machine it's `nativeBuildInputs`. If it's linked as a library or its path is recorded to be executed later in the host environment, it's `buildInputs`

Reply:
Not quite accurate. `nativeBuildInputs` and `buildInputs` is more related to cross compilation use-cases. For example, if you want to compile an aarch64 package on an x86_64 machine, you will need build-tools that are native to the build platform (hence `nativeBuildInputs`), while the build itself is going to need libraries and other dependencies of the host platform (host platform being the platform we are building for).

A lot more detail is provided in the Nixpkgs manual: https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-reference

Reply:
I think we're saying the same thing:)